app/code/Magento/Catalog/Model/Indexer/Category/Flat/AbstractAction.php

```diff
@@ -77,20 +76,31 @@ class AbstractAction
      */
     protected $skipStaticColumns = [];
 
+    /**
+     * @var SkipStaticColumnsProvider
+     */
+    private $staticColumnsProvider;
+
     /**
      * @param ResourceConnection $resource
      * @param StoreManagerInterface $storeManager
      * @param Helper $resourceHelper
+     * @param MetadataPool|null $categoryMetadata
+     * @param SkipStaticColumnsProvider|null $provider
```

Comment:
Please name the variables according to the class name

```suggestion
     * @param MetadataPool|null $metadataPool
     * @param SkipStaticColumnsProvider|null $skipStaticColumnsProvider
```